processor/src/main/java/org/mapstruct/ap/internal/model/ContainerMappingMethod.java

```diff
@@ -36,11 +36,8 @@
  *
  * @author Filip Hrisafov
  */
-public abstract class ContainerMappingMethod extends MappingMethod {
```

Comment:
> I completely understand you not liking the long inheritance chain. Do you perhaps have some other suggestion?

I think we should park this one. But on thing we might consider is to use composition instead of inheritance. Or a combination: we still have a base class `MappingMethod` and top level methods like `BeanMappingMethod`, `MapMappingMethod` but we compose other behavior out of the `ContainerMappingMethod` and ` NormalTypeMappingMethod`